bin/check-changelog.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Given a set of file paths as arguments, checks whether the file paths also
+ * include a requisite CHANGELOG entry. The file paths should include packages
+ * maintained with a CHANGELOG at, e.g. `packages/i18n/CHANGELOG.md`.
+ *
+ * Example (Error):
+ *
+ * ```
+ * node bin/check-changelog.js /Gutenberg/packages/i18n/src/index.js
+ * ```
+ *
+ * Example (Success):
+ *
+ * ```
+ * node bin/check-changelog.js /Gutenberg/packages/i18n/src/index.js \
+ *  /Gutenberg/packages/i18n/CHANGELOG.md
+ * ```
+ */
+
+/**
+ * External dependencies
+ */
+
+const { relative, sep, join } = require( 'path' );
+const minimatch = require( 'minimatch' );
+
+/**
+ * Internal dependencies
+ */
+
+const { ignoreChanges = [] } = require( '../lerna' );
+
+/**
+ * Returns true if the given path is ignored by the Lerna `ignoreChanges`
+ * configuration, or false otherwise.
+ *
+ * @param {string} path Path to test.
+ *
+ * @return {boolean} Whether path is ignored.
+ */
+function isIgnored( path ) {
+	return ignoreChanges.some( ( pattern ) => minimatch( path, pattern ) );
+}
+
+/**
+ * An array of file paths relative the `packages` directory.
+ *
+ * @type {string[]}
+ */
+const paths = process.argv
+	.slice( 2 )
+	.map( ( path ) => relative( 'packages', path ) );
+
+/**
+ * The set of verified package names, i.e. those which have been confirmed to
+ * include an associated CHANGELOG revision.
+ *
+ * @type {Set}
+ */
+const verified = new Set;
```

Comment:
Is it okey to omit parentheses?

Reply:
>Is it okey to omit parentheses?

Yes:

>`new Foo` is equivalent to `new Foo()`, i.e. if no argument list is specified, `Foo` is called without arguments.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new#Description

Though some people might argue it's better to just be consistent:

https://eslint.org/docs/rules/new-parens

Reply:
Interesting. I'm fine to leave it as is if JS is happy :)